### DISCUSSION TITLE
app in swarm mode without internet access

### DISCUSSION THREAD

**Discussion by erickythierry** [caprover/caprover#2152] [Q&A] [answered]
Greetings.
I did a test configuration using 2 Oracle Linux servers in cluster mode.
Server 1 being the master and Server 2 being the worker
I have a project that uses Firebase for DB. If I deploy it to the master server it works perfectly, but if I change the app's node ID so it runs on the worker server, it can't access the Firebase servers.

I did the worker mode configuration process using the alternative method in the Caprover panel

![photo_2024-09-25_18-50-58](https://github.com/user-attachments/assets/2fa4ec4e-3aa4-446f-b0f7-7eaa539d76eb)

Is there any swarm configuration I can do to fix this issue?
`@firebase/database: FIREBASE WARNING: {"code":"app/invalid-credential","message":"Credential implementation provided to initializeApp() via the \"credential\" property failed to fetch a valid Google OAuth2 access token with the following error: \"Error fetching access token: Error while making request: getaddrinfo EAI_AGAIN accounts.google.com. Error code: EAI_AGAIN\"."}
`

**Comment by erickythierry** [comment]
![image](https://github.com/user-attachments/assets/3c7f9d6c-131f-4bbe-ada6-0d0684d7175e)

I made this simple code and deployed it to the worker to test and there really is no internet connection

![image](https://github.com/user-attachments/assets/f7cdae61-e9a6-48f4-918e-e08e4719ab06)

If anyone can give me any insight into this problem, I would appreciate it.

**Reply by githubsaturn** [reply, marked as answer]
This is not a CapRover issue. You should contact your VPS provider to see what's causing this. Probably a custom DNS?

**Reply by erickythierry** [reply]
That was it. I changed from Oracle Linux to Ubuntu and it solved the problem. Sorry for opening this question.😅
